Open the app and save supported models

shavit opened the pull request:
Supported model files will open the app and be saved. 

Closes #35 
Closes https://github.com/psugihara/FreeChat/issues/49 (with https://github.com/psugihara/FreeChat/pull/46/commits/757f50c59106e1da02eb983c06d9b702f80fbb81)

shavit commented:
I edited `project.pbxproj` directly to put back the development team, and also formatted the new file with 2 spaces.

psugihara commented on mac/FreeChat/FreeChatAppDelegate.swift:
```diff
@@ -0,0 +1,30 @@
+      let existingModels = try viewContext.fetch(req).compactMap({ $0.url })
+
+      for url in urls {
+        guard !existingModels.contains(url) else { continue }
```
This guard is not working correctly, re-adding the same model adds it to the list again.

psugihara commented on mac/FreeChat/FreeChatAppDelegate.swift:
```diff
@@ -0,0 +1,30 @@
+      for url in urls {
+        guard !existingModels.contains(url) else { continue }
+        let insertedModel = try Model.create(context: viewContext, fileURL: url)
+        selectedModelId = insertedModel.id?.uuidString
```
Not sure why but this is not changing the selected model in the dropdown (see video).

shavit replied:
Maybe because `selectedModelId` is being initialized with a default value. I addressed this and more issues that I found in https://github.com/psugihara/FreeChat/pull/47

psugihara reviewed (changes requested):
Awesome work, this is close. Noted a few things inline, here's a video:


https://github.com/psugihara/FreeChat/assets/282016/f4927bff-758c-42c9-a2bc-deceb7f534ed

psugihara commented:
One more thing that seems broken (not sure if it was this or the last PR). The size attribute is no longer being populated for models. Here's a model I added with the TestFlight build, the others were added with this build. Note that the size is displayed because it is present on the Core Data Model.

<img width="522" alt="Screenshot 2024-01-13 at 2 14 21 PM" src="https://github.com/psugihara/FreeChat/assets/282016/b282fa5c-e678-431f-8b9a-e27768fb6556">

shavit commented:
<img width="607" alt="Screenshot 2024-01-13 at 5 21 49 PM" src="https://github.com/psugihara/FreeChat/assets/252122/aaff8124-7817-4507-bd4b-6430ba11abf7">
<img width="422" alt="Screenshot 2024-01-13 at 5 22 04 PM" src="https://github.com/psugihara/FreeChat/assets/252122/01bcf0bc-e15e-457f-9871-2acc231da220">

Here I deleted few models and there is no selection. I also moved and renamed the files to see how it behaves.

shavit reviewed (commented):
Pull request https://github.com/psugihara/FreeChat/pull/47 need to precede this one

psugihara commented:
hell yes, this works great!! thank you @shavit
